packages/babel-generator/src/printer.ts

```diff
@@ -929,6 +927,11 @@ class Printer {
 
     let val;
     if (isBlockComment) {
+      if (this._noLineTerminator && NEWLINE.test(comment.value)) {
+        this._printedComments.delete(comment);
+        return;
+      }
+
```

Comment:
Can we move this to L910? This merges the addition logic.

Reply:
I moved it even earlier, before the `shouldPrintComment` check.

Reply:
I personally prefer to put it after `isBlockComment` to avoid detection of line comments, but the actual impact should be small.

Reply:
Since `shouldPrintComment` is an user-provided function, I think we should only call it if we know for sure that we are going to print the comment.